Src/FluentAssertions/Data/IDataEquivalencyAssertionOptions.cs

```diff
@@ -0,0 +1,183 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq.Expressions;
+
+using FluentAssertions.Equivalency;
+
+namespace FluentAssertions.Data
+{
+    /// <summary>
+    /// Provides access to configuration for equivalency assertions on System.Data types (<see cref="DataSet"/>,
+    /// <see cref="DataTable"/>, <see cref="DataRow"/>, <see cref="DataColumn"/>, <see cref="DataRelation"/>,
+    /// <see cref="Constraint"/>).
+    /// </summary>
+    /// <typeparam name="T">The System.Data type being tested for equivalency.</typeparam>
+    public interface IDataEquivalencyAssertionOptions<T> : IEquivalencyAssertionOptions
+    {
+        /// <summary>
+        /// Specifies that the subject and the expectation should not be considered non-equivalent if their exact data types do not match.
+        /// </summary>
+        IDataEquivalencyAssertionOptions<T> AllowingMismatchedTypes();
+
+        /// <summary>
+        /// Specifies that when comparing <see cref="DataTable.Columns"/>, columns that are unmatched between the subject and the expectation should be ignored.
+        /// </summary>
+        IDataEquivalencyAssertionOptions<T> IgnoringUnmatchedColumns();
+
+        /// <summary>
+        /// Specifies the <see cref="RowMatchMode"/> that should be used when comparing <see cref="DataTable.Rows"/>. By default, rows are matched by their index in the <see cref="DataTable.Rows"/> collection. But, if the <see cref="DataTable"/> has a <see cref="DataTable.PrimaryKey"/> set, it is possible to use <see cref="RowMatchMode.PrimaryKey"/> to indicate that rows should be matched by their primary key values, irrespective of their index within the <see cref="DataTable.Rows"/> collection.
+        /// </summary>
+        /// <param name="newRowMatchMode">The <see cref="RowMatchMode"/> to use when comparing <see cref="DataTable.Rows"/> between the subject and the expectation.</param>
+        IDataEquivalencyAssertionOptions<T> UsingRowMatchMode(RowMatchMode newRowMatchMode);
```

Comment:
I think the reason I put it there is because if the parameter is named `rowMatchMode`, then it matches the name of the field, so this particular configuration method is forced to refer to the field with `this.` qualifier, while none of the others are, and the inconsistency doesn't look nice:
```
        public IDataEquivalencyAssertionOptions<T> AllowingMismatchedTypes()
        {
            allowMismatchedTypes = true;
            return this;
        }

        public IDataEquivalencyAssertionOptions<T> IgnoringUnmatchedColumns()
        {
            ignoreUnmatchedColumns = true;
            return this;
        }

        public IDataEquivalencyAssertionOptions<T> UsingRowMatchMode(RowMatchMode rowMatchMode)
        {
            this.rowMatchMode = rowMatchMode;
            return this;
        }

        public IDataEquivalencyAssertionOptions<T> ExcludingOriginalData()
        {
            excludeOriginalData = true;
            return this;
        }
```
What is preferred? With `this.` qualifier? Or, with some other name for the parameter? Or is `new` an acceptable way to avoid having to qualify the field name in the assignment??

Reply:
I would rather have `this` qualifier in the implementation than a prefix in the API.

Reply:
Actually, I've just noticed that the implementation is `internal`. Would it be okay for the parameter name to differ between the public interface and the implementation, so that the approved API surface does not have the prefix, and the implementation does have it to avoid needing `this.`?